File: LeetCode/1880. Check if Word Equals Summation of Two Words/solution.cpp
/**
 * Runtime: 4 ms
 * Memory Usage: 5.8 MB 
 */
class Solution {
public:
    int numerical(const string &word) {
        int num = 0;
        for (auto &c: word) {
            num *= 10;
            num += c - 'a';
        }
        return num;
    }

    bool isSumEqual(string firstWord, string secondWord, string targetWord) {
        return numerical(firstWord) + numerical(secondWord) == numerical(targetWord);
    }
};

File: LeetCode/1880. Check if Word Equals Summation of Two Words/solution.py
"""
Runtime: 28 ms
Memory Usage: 14.8 MB
"""
class Solution:
    def isSumEqual(self, firstWord: str, secondWord: str, targetWord: str) -> bool:
        def numerical(word):
            ret = 0
            for c in word:
                ret *= 10
                ret += ord(c) - ord('a') 
            return ret
        return numerical(firstWord) + numerical(secondWord) == numerical(targetWord)  
